medusalib/src/main/java/com/trendyol/medusalib/navigator/MultipleStackNavigator.kt

```diff
@@ -274,6 +305,28 @@ open class MultipleStackNavigator(
         return true
     }
 
+    private fun notifyFragmentDestinationChange(fragment: Fragment) {
+        fragment.lifecycle.addObserver(object: DefaultLifecycleObserver {
+            override fun onStart(owner: LifecycleOwner) {
+                owner.lifecycle.removeObserver(this)
+                fragment.viewLifecycleOwner.lifecycle.addObserver(
+                    object : DefaultLifecycleObserver {
+                        override fun onCreate(owner: LifecycleOwner) {
+                            destinationChangeLiveData.value = fragment
+                        }
+
+                        override fun onDestroy(owner: LifecycleOwner) {
+                            if (destinationChangeLiveData.value == fragment) {
+                                destinationChangeLiveData.value = null
+                            }
+                            owner.lifecycle.removeObserver(this)
+                        }
+                    }
+                )
+            }
+        })
+    }
```

Comment:
Do we have to observe fragment's lifecycle? We can also directly observe the `fragment.viewLifecycleOwner`, like:

```kotlin
    private fun notifyFragmentDestinationChange(fragment: Fragment) {
        fragment.viewLifecycleOwner.lifecycle.addObserver(
            object : DefaultLifecycleObserver {
                override fun onCreate(owner: LifecycleOwner) {
                    destinationChangeLiveData.value = fragment
                }

                override fun onDestroy(owner: LifecycleOwner) {
                    if (destinationChangeLiveData.value == fragment) {
                        destinationChangeLiveData.value = null
                    }
                    owner.lifecycle.removeObserver(this)
                }
            }
        )
    }
```